### PR TITLE
fix: images can be configured with properties

### LINKED ISSUE #296: PropertyConfigHandler doesn't seem to be used - in Gradle
### Description
[PropertyConfigHandler](https://github.com/eclipse/jkube/blob/cd475427f59c7df731e57b0516d4e8df0a0e9c3a/jkube-kit/build/service/docker/src/main/java/org/eclipse/jkube/kit/build/service/docker/config/handler/property/PropertyConfigHandler.java#L153) seems to be completely ignored, same as [DockerComposeConfigHandler](https://github.com/eclipse/jkube/blob/cd475427f59c7df731e57b0516d4e8df0a0e9c3a/jkube-kit/build/service/docker/src/main/java/org/eclipse/jkube/kit/build/service/docker/config/handler/compose/DockerComposeConfigHandler.java#L43).

Functionality in ImageConfigResolver#resolve seems to be also something useless at the moment: https://github.com/eclipse/jkube/blob/9db93fb42d1dcade59278ce8f1b433b869e0927d/jkube-kit/build/service/docker/src/main/java/org/eclipse/jkube/kit/build/service/docker/config/handler/ImageConfigResolver.java#L72-L87

Apparently this is something that's already advised/warned in the docs:

https://github.com/eclipse/jkube/blob/9db93fb42d1dcade59278ce8f1b433b869e0927d/kubernetes-maven-plugin/doc/src/main/asciidoc/inc/_introduction.adoc#L215-L218
image

But the code responsible for this has been included in the project nonetheless.

We need to reevaluate if this code+functionality is relevant in the project and act accordingly.

### PR DESCRIPTION
**Pull request by manusa**
## Description

Fixes #296

fix: images can be configured with properties

## Type of change

 - [x] Bug fix (non-breaking change which fixes an issue)
 - [x] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [ ] I have read the [contributing guidelines](https://www.eclipse.dev/jkube/contributing)
 - [ ] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [ ] My code follows the style guidelines of this project
 - [ ] I Keep It Small and Simple: The smaller the PR is, the easier it is to review and have it merged
 - [ ] I use [conventional commits](https://www.conventionalcommits.org/) in my commit messages
 - [ ] I have performed a self-review of my code
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [ ] I have added tests that prove my fix is effective or that my feature works
 - [ ] New and existing unit tests pass locally with my changes
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Comment by manusa** [comment]
### Eclipse JKube [CI Report](https://github.com/eclipse-jkube/ci)

Started new GH workflow run for https://github.com/eclipse-jkube/jkube/pull/3042 (_2024-05-10T14:06:08Z_)

:gear: [JKube E2E Tests (9033428738)](https://github.com/eclipse-jkube/ci/actions/runs/9033428738) :hourglass_flowing_sand:
- :hourglass_flowing_sand: [Bootstrap CI test run (#3042)](https://github.com/eclipse-jkube/ci/actions/runs/9033428738/job/24823759405)

File: gradle-plugin/it/src/it/extension-configuration/expected/expected-config.yml [hunk 1]
```diff
@@ -65,20 +65,16 @@ images:
 - name: "registry/image:tag"
   build:
     contextDir: "@endsWith('extension-configuration/src/main/context-dir')@"
-    dockerFile: "@endsWith('extension-configuration/src/main/context-dir/Dockerfile')@"
```
Comment (manusa) [inline gradle-plugin/it/src/it/extension-configuration/expected/expected-config.yml:68]: Previously, the images that were declared in the context of the KubernetesExtension class were mutated by the DefaultGeneratorManager.

In the current implementation, (unresolved) images passed in to the DefaultGeneratorManager are no longer mutated and are cloned and enriched incrementally instead.

This is reflected here, where the expected config should only show the values passed in by the user in the build.gradle configuration.

File: jkube-kit/doc/src/main/asciidoc/inc/build/_configuration.adoc [hunk 2]
```diff
@@ -3,39 +3,46 @@ of an image configuration. The following configuration options are supported:
...
 .Build configuration (<<config-image, image>>)
-[cols="1,5"]
+[cols="1,5,1"]
 |===
-| Element | Description
+| Element | Description | Property
 
 | <<config-image-build-assembly, *assembly*>>
 | Specifies the assembly configuration as described in <<config-image-build-assembly,Build Assembly>>
+| `jkube.container-image.assembly.xxx`
```
Comment (rohanKanojia) [inline jkube-kit/doc/src/main/asciidoc/inc/build/_configuration.adoc:12]: Is this correct?

Comment (rohanKanojia) [reply jkube-kit/doc/src/main/asciidoc/inc/build/_configuration.adoc:12]: okay, it seems to be the latter.

File: jkube-kit/doc/src/main/asciidoc/inc/build/_assembly.adoc [hunk 3]
```diff
@@ -6,35 +6,41 @@ logic (see <<jkube-plugin, JKube Plugin>> for more details).
...
   <<build-assembly-layer, Assembly - Layer>> below.
+|
 
 | *exportTargetDir*
 | Specification whether the `targetDir` should be exported as a volume. This value is `true` by default except in the
   case the `targetDir` is set to the container root (`/`). It is also `false` by default when a base image is used with
   `from` since exporting makes no sense in this case and will waste disk space unnecessarily.
+| `jkube.container-image.assembly.exportTargetDir`
 
 | *excludeFinalOutputArtifact*
```
Comment (rohanKanojia) [inline jkube-kit/doc/src/main/asciidoc/inc/build/_assembly.adoc:40]: why is property not added for this field?

File: jkube-kit/build/api/src/main/java/org/eclipse/jkube/kit/build/api/config/property/PropertyConfigResolver.java [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+  }
+
+  private WatchImageConfiguration extractWatchConfig(ImageConfiguration fromConfig, ValueProvider valueProvider) {
+    final WatchImageConfiguration config = fromConfig.getWatchConfiguration() != null ?
+      fromConfig.getWatchConfiguration() : new WatchImageConfiguration();
+    return config.toBuilder()
+      .interval(valueProvider.getInteger(ConfigKey.WATCH_INTERVAL, config.getIntervalRaw()))
+      .postGoal(valueProvider.getString(ConfigKey.WATCH_POSTGOAL, config.getPostGoal()))
+      .postExec(valueProvider.getString(ConfigKey.WATCH_POSTEXEC, config.getPostExec()))
+      .modeString(valueProvider.getString(ConfigKey.WATCH_POSTGOAL, config.getMode() == null ? null : config.getMode().name()))
```
Comment (rohanKanojia) [inline jkube-kit/build/api/src/main/java/org/eclipse/jkube/kit/build/api/config/property/PropertyConfigResolver.java:158]: Shouldn't we be using ConfigKey.WATCH_MODE here ?

File: jkube-kit/doc/src/main/asciidoc/inc/build/_assembly.adoc (hunk 3 above)
Comment (manusa) [reply jkube-kit/doc/src/main/asciidoc/inc/build/_assembly.adoc:40]: It was not part of the original list of properties. We can certainly add it (Maybe as a follow up)

**Review by rohanKanojia — APPROVED** [review]

File: jkube-kit/build/api/src/main/java/org/eclipse/jkube/kit/build/api/config/property/PropertyConfigResolver.java (hunk 4 above)
Comment (manusa) [reply jkube-kit/build/api/src/main/java/org/eclipse/jkube/kit/build/api/config/property/PropertyConfigResolver.java:158]: Definitely, this seems to have been carried over from DMP:
https://github.com/fabric8io/docker-maven-plugin/blob/25e772ed663d34bf2932eb48baabd6492065b0cf/src/main/java/io/fabric8/maven/docker/config/handler/property/PropertyConfigHandler.java#L458

That should be fixed there too.

I'll change here as soon as you finish the review so I can push all the changes at once.

**Pull request merged**